src/cuda/device_conj.cu

```diff
@@ -0,0 +1,72 @@
+#include "blas/device.hh"
+#include "thrust/complex.h"
+
+#if defined(BLAS_HAVE_CUBLAS)
+
+namespace blas {
+
+template <typename scalar_t>
+__global__ void conj_kernel(
+    int64_t n,
+    scalar_t* src, int64_t inc_src,
+    scalar_t* dst, int64_t inc_dst)
+{
+    using thrust::conj;
+
+    int i = blockIdx.x * blockDim.x + threadIdx.x;
+    if (i < n)
+        dst[ i*inc_dst ] = conj( src[ i*inc_src ] );
+}
+
+//------------------------------------------------------------------------------
+/// Conjugates each element of the vector src and stores in dst.
+///
+/// @param[in] n
+///     Number of elements in the vector. n >= 0.
+///
+/// @param[in] src
+///     Pointer to the input vector of length n.
+///
+/// @param[in] inc_src
+///     Stride between elements of src. inc_src >= 1.
+///
+/// @param[out] dst
+///     Pointer to output vector
+///     On exit, each element dst[i] is updated as dst[i] = conj( src[i] ).
+//      dst may be the same as src.
+///
+/// @param[in] inc_dst
+///     Stride between elements of dst. inc_dst >= 1.
+///
+/// @param[in] queue
+///     BLAS++ queue to execute in.
+///
+template <typename scalar_t>
+void conj(
+    int64_t n,
+    scalar_t* src, int64_t inc_src,
```

Comment:
Typically, src would be `const`. Our convention is to put `const` after the type, so types are read correctly from right-to-left: `scalar_t const*` is "pointer to constant scalar_t".